pkg/controller/awsprivatelink/awsprivatelink_controller_test.go

```diff
@@ -1484,6 +1484,24 @@ users:
 				cd.Status.Conditions[i].LastProbeTime = metav1.Time{}
 				cd.Status.Conditions[i].LastTransitionTime = metav1.Time{}
 			}
+			if len(test.expectedConditions) < len(clusterDeploymentAWSPrivateLinkConditions) {
```

Comment:
This logic feels like it might catch someone off guard in the future. If I'm reading this right we're saying that if the test has fewer expected conditions than the two we know should be there, make sure the two we know should be there are injected. However if the test author was expecting two different conditions, this wouldn't trigger. It's also just strange that we're passing in a set of expected conditions and then changing it under the covers on the test author. 

A couple options:

  1. Add the expected conditions in every test case.
  2. Change the assumption that we're testing the ENTIRE array of conditions to verifying that the expected conditions are present and set as we expect. Honestly this exercise does show how brittle these tests are, whereas arguably we should unit test that condition A we expect is in state X. Additional conditions could be ignored.

I could go either way, I think for a small simple controller like this I'd probably do option (1) as we're really just checking two conditions here. But in controllers with a large number of possible conditions, we may not want to define every condition for every test, and then (2) would make more sense.

Reply:
I kinda took bits and pieces of both suggestions and implemented them as fit.
Originally I moved the `assertConditionStatus` and my new implementation of `assertConditions` to `pkg\test\clusterdeployment\`, however I ended up creating a cyclic dependency via controller utils. So I have added them individually to the controllers for now. Let me know if you're okay with my current implementation and if there's an another place we could move those methods to without creating a cyclic dependency